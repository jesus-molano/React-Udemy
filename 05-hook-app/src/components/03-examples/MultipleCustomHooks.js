import useCounter from '../../hooks/useCounter';
import useFetch from '../../hooks/useFetch';
import './customHooks.css';

const MultipleCustomHooks = () => {
  const { counter, increment, decrement } = useCounter(1);
  const url = `https://www.breakingbadapi.com/api/quotes/${counter}`;

  const { loading, data } = useFetch(url);
  const {author, quote} = (!loading && data.length > 0) && data[0]

  return (
    <div>
      <h1>BreakingBad Quotes</h1>
      <hr />
      {loading && <div className="alert alert-info text-center">Loading...</div>}
      {!loading && (
          <blockquote className="blockquote text-end">
            <p className="mb-3">{quote}</p>
            <footer className="blockquote-footer">{author}</footer>
          </blockquote>
        )
      }
      <button className="btn btn-success" onClick={decrement}>Prev Quote</button>
      <button className="btn btn-success" onClick={increment}>Next Quote</button>
    </div>
  )
}

export default MultipleCustomHooks
