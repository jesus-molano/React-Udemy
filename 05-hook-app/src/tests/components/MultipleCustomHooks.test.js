import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../components/03-examples/MultipleCustomHooks';
import useCounter from '../../hooks/useCounter';
import useFetch from '../../hooks/useFetch';
jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCounter');

describe('<MultipleCustomHooks /> tests', () => {
	beforeEach(() => {
		useCounter.mockReturnValue({
			counter: 10,
			increment: () => {},
			decrement: () => {},
			reset: () => {},
		});
	});

	test('should show correctly', () => {
		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null,
		});

		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper).toMatchSnapshot();
	});

	test('should show the info', () => {
		useFetch.mockReturnValue({
			data: [
				{
					author: 'jesus',
					quote: 'Esto es una prueba',
				},
			],
			loading: false,
			error: null,
		});
		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper.find('.alert').exists()).toBe(false);
		expect(wrapper.find('.mb-3').text().trim()).toBe('Esto es una prueba');
		expect(wrapper.find('footer').text().trim()).toBe('jesus');
	});
});
