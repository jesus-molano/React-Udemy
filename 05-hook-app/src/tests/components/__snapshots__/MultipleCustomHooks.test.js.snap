// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultipleCustomHooks /> tests should show correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Prev Quote
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
    Next Quote
  </button>
</div>
`;
